backend/src/routes/metadata-routes.ts

```diff
@@ -92,13 +92,16 @@ const metadataRoutes: FastifyPluginAsync = async (server) => {
       const signer = new Wallet(privateKey, provider)
       const sponsorWalletBalance = await signer.getBalance();
       const sponsorAddress = await signer.getAddress();
+      let sponsorBalance = 0;
 
-      //get native balance of the sponsor in the EtherSpotPaymaster-contract
-      const paymasterContract = new Contract(networkConfig.contracts.etherspotPaymasterAddress, EtherspotAbi.default, provider);
-      const sponsorBalance = await paymasterContract.getSponsorBalance(sponsorAddress);
+      if (networkConfig.contracts.etherspotPaymasterAddress) {
+        //get native balance of the sponsor in the EtherSpotPaymaster-contract
+        const paymasterContract = new Contract(networkConfig.contracts.etherspotPaymasterAddress, EtherspotAbi.default, provider);
+        sponsorBalance = await paymasterContract.getDeposit();
```

Comment:
What if this fails or times-out? Pls handle it.

Reply:
have added try-catch block